src/Chapter3.hs

```diff
@@ -1006,6 +1170,143 @@ Implement data types and typeclasses, describing such a battle between two
 contestants, and write a function that decides the outcome of a fight!
 -}
 
+data Action
+  = Attack
+  | DrinkPotion
+  | CastSpell
+  | RunAway deriving (Show)
+
+data Monster = Monster
+  { monsterHealth  :: Int
+  , monsterAttack  :: Int
+  , monsterActions :: [Action]
```

Comment:
By the task defintion, `Monster` can't cast spells or drink potions. So it would be better to have two separate types `KnightAction` and `MonsterAction` to avoid having invalid states